### PR TITLE
Migrate tests in `view` test package to `JUnit 5` using `DatabindTestUtil`

### PR DESCRIPTION
**Pull request by JooHyukKim**
As per title and https://github.com/FasterXML/jackson-databind/discussions/4190

### CODE REVIEW

**Review by pjfanning — CHANGES_REQUESTED** [review]
jupiter and junit - the asserts are reversed - in junit, expected comes first, in jupiter, expected comes 2nd.

I am against using jupiter as a result. I prefer the tests and the messages to be correct. Jupiter is not that great. Junit4 works fine. It is not worth the hassle. Just busy work - having to rewrite every assert. And then getting people to review the changes. And in the end, the test was fine before the work even started.

**Comment by pjfanning** [comment]
https://junit.org/junit5/docs/snapshot/user-guide/#migrating-from-junit4-tips suggests that when using jupiter (junit 5), you can still import the asserts from `org.junit.Assert` - in this case, you wouldn't need to change the param order.

File: src/test/java/com/fasterxml/jackson/databind/BaseMapTest.java [hunk 1]
```diff
@@ -292,9 +292,18 @@ protected void assertStandardEquals(Object o)
...
         throws IOException
+    {
+        return writeAndMap2(m, value);
+    }
+
+    /**
+     * To prepare for 3.0 that will move away from JUnit 4, we need to avoid using protected methods.
+     */
+    @SuppressWarnings("unchecked")
+    public static Map<String,Object> writeAndMap2(ObjectMapper m, Object value)
```
Comment (cowtowncoder) [inline src/test/java/com/fasterxml/jackson/databind/BaseMapTest.java:305]: This works, but I'd also recommend also checking if we could and should just get rid of this method.
I _think_ it's old one and predates existence of `writeValueAsString()` so was more useful before that.

Comment (cowtowncoder) [reply src/test/java/com/fasterxml/jackson/databind/BaseMapTest.java:305]: Ok, that's called from quite a few places. Hmmh.

Also: for new, non-inheritance based test classes, maybe we should migrate methods into `TestUtil` class with static methods, and essentially start getting rid of `BaseXxx` classes for tests.
During transition `BaseMapTest` could just call methods in `TestUtil` (or maybe it'd be `DatabindTestUtil`?)

**Comment by JooHyukKim** [comment]
> https://junit.org/junit5/docs/snapshot/user-guide/#migrating-from-junit4-tips suggests that when using jupiter (junit 5), you can still import the asserts from `org.junit.Assert` - in this case, you wouldn't need to change the param order.

Thank you for the tips! Really helps. The thing is tho, wrt using `org.junit.Assert` in JUnit 5 tests, gives this warning.
image

**Comment by JooHyukKim** [comment]
> https://junit.org/junit5/docs/snapshot/user-guide/#migrating-from-junit4-tips suggests that when using jupiter (junit 5), you can still import the asserts from `org.junit.Assert` - in this case, you wouldn't need to change the param order.

Thank you for the tips! Really helps. The thing is tho, wrt using `org.junit.Assert` in JUnit 5 tests, gives this warning.

image

EDITED: nvm warning is not worth flpping params.

**Comment by JooHyukKim** [comment]
> I am against using jupiter as a result. I prefer the tests and the messages to be correct. Jupiter is not that great. Junit4 works fine. It is not worth the hassle. Just busy work - having to rewrite every assert. And then getting people to review the changes. And in the end, the test was fine before the work even started.

I concur, Jupiter is not that much different in terms of functionality or usage itself. What we are aiming for is eventual (in Jackson 3) consistency among test cases 🤔 --if I understood right.

File: src/test/java/com/fasterxml/jackson/databind/BaseMapTest.java (hunk 1 above)
Comment (JooHyukKim) [reply src/test/java/com/fasterxml/jackson/databind/BaseMapTest.java:305]: Sounds great, added so! 👍🏼

**Comment by cowtowncoder** [comment]
@pjfanning WDYT? I guess I the big question really is, wrt #4190: what is the ROI. To me JUnit 5 has some nice incremental changes (assertThrows as an example), but I wouldn't want lots of time being spent to get minor benefits.

And sort of related, I have sort of grown to like AssertJ, now used by both `jackson-core` and `jackson-databind`. It offers bit different way for doing assertions.

So: although we can definitely do conversion for this package, more important question is whether to continue converting things, either actively (like here) -- or reactively, like, when we change tests anyway, upgrade as part of other changes.
To me that might be better way to go: when rewriting things, then upgrade.

Because I also think that over time we should migrate to later versions of things we use.

**Comment by JooHyukKim** [comment]
Hmmm, good question @cowtowncoder , for what benefit? 🤔 The main factor I considered signficant was consistency among tests, but now I am wondering if there was any inconsistency was getting in our way to start with --I guess not. 

Considering we also start using `AssertJ`, there will be a few different ways test classes/methods will look. But I don't see no harm in that either.

Wrt `active" vs `reactive` upgrade, I'm a bit on "planned active", because reactive direction will take up more overall energy spent.

**Comment by cowtowncoder** [comment]
@JooHyukKim On active vs reactive... to me active one means spending lots of up-front effort to change tests that are working just fine. Unless conversion is automated or such. Instead of working on other things that are more urgent. But it may be true that converting test classes when doing smaller unrelated changes is impractical.

Anyway. I think I am still +1 on conversion, but doing it slowly.
So feel free to re-open this issue. I think I can have another look and then probably merge it.

**Comment by JooHyukKim** [comment]
Hmm... Thanks you for sharing, @cowtowncoder 👍🏼 I concur, reactive conversion is more practical.

Maybe, I can try to to take on the conversion job weekly or bi-weekly, whichever suits your case(time-wise)?

I will see what we can do with [OpenRewrite](https://docs.openrewrite.org/running-recipes/popular-recipe-guides/migrate-from-junit-4-to-junit-5) suggested in [earlier discussion](https://github.com/FasterXML/jackson-databind/discussions/4190#discussioncomment-7484721)

This PR now seems viable. One benefit is around the `DatabindTestUtil` you suggested. Migrated JUnit 5 tests' no longer reference `BaseMapTest`, thus resolving the overlap with test classes with previous JUnit versions

**Review by cowtowncoder — APPROVED** [review]

**Comment by cowtowncoder** [comment]
Ok yeah, package per week or something works. Merging to `master` is the laborious part.

On imports: I am fine ordering `static` imports before non-static, but I think they should follow similar ordering as non-static before -- from more general to more specific:

1. First, JDK platform types (`java.io`, `java.util` etc)
2. General 3rd party types (`org.junit`)
3. Jackson core types: annotations, core, databind (in this order)
4. (for non-core Jackson components) Component-specific types

**Pull request merged**